src/com/palmergames/bukkit/towny/event/resident/ResidentToggleModeEvent.java

```diff
@@ -0,0 +1,63 @@
+package com.palmergames.bukkit.towny.event.resident;
+
+import com.palmergames.bukkit.towny.object.Resident;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class ResidentToggleModeEvent extends Event implements Cancellable {
+	private static final HandlerList HANDLER_LIST = new HandlerList();
+	private boolean cancelled;
+	private String cancelMessage = "";
+	private final Resident resident;
+	private final String mode;
+	private final boolean toggleOn;
+	
+	public ResidentToggleModeEvent(Resident resident, String mode, boolean toggleOn) {
```

Comment:
I think the way the constructor is works we don't need to supply a toggleOn variable.